src/libraries/System.Private.CoreLib/src/System/Threading/Interlocked.cs

```diff
@@ -69,6 +69,24 @@ public static partial class Interlocked
         public static ulong Exchange(ref ulong location1, ulong value) =>
             (ulong)Exchange(ref Unsafe.As<ulong, long>(ref location1), (long)value);
 
+        /// <summary>Sets a single-precision floating point number to a specified value and returns the original value, as an atomic operation.</summary>
+        /// <param name="location1">The variable to set to the specified value.</param>
+        /// <param name="value">The value to which the <paramref name="location1"/> parameter is set.</param>
+        /// <returns>The original value of <paramref name="location1"/>.</returns>
+        /// <exception cref="NullReferenceException">The address of location1 is a null pointer.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static float Exchange(ref float location1, float value)
+            => Unsafe.BitCast<int, float>(Exchange(ref Unsafe.As<float, int>(ref location1), Unsafe.BitCast<float, int>(value)));
```

Comment:
I do not have an opinion. What do you think?

Generic instantiations of small leaf methods parameterized by concrete types are small fixed cost. I am not worried about those.

Generic instantiations parameterized by generic parameters or generic instantiations of large generic structure is what I am looking for as red flags. These are multiplying costs and/or large fixed costs.

Reply:
`SingleToInt32Bits` uses BitCast in its managed body, but it's intrinsic itself. I'm not sure how common `Interlocked` is compared to `Unsafe`.

Is the cost using `BitConverter` definitely smaller? Is there cost for loading a new type? `Interlocked` is mainly using `Unsafe`.

Reply:
BitConverter is relatively popular type. It is likely that it will be loaded by something in the application. On the other hand, Interlocked operations on float and doubles are not very common. 

The difference between the two options is negligible. You can choose either based on your personal style preference here.